server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/identity/PostgresCustomIdentityDAO.java

```diff
@@ -0,0 +1,233 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.postgres.identity;
+
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.BCC;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.EMAIL;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.HTML_SIGNATURE;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.ID;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.MAY_DELETE;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.NAME;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.REPLY_TO;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.SORT_ORDER;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.TABLE_NAME;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.TEXT_SIGNATURE;
+import static org.apache.james.jmap.postgres.identity.PostgresCustomIdentityModule.PostgresCustomIdentityTable.USERNAME;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+import javax.inject.Inject;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.Username;
+import org.apache.james.jmap.api.identity.CustomIdentityDAO;
+import org.apache.james.jmap.api.identity.IdentityCreationRequest;
+import org.apache.james.jmap.api.identity.IdentityNotFoundException;
+import org.apache.james.jmap.api.identity.IdentityUpdate;
+import org.apache.james.jmap.api.model.EmailAddress;
+import org.apache.james.jmap.api.model.Identity;
+import org.apache.james.jmap.api.model.IdentityId;
+import org.jooq.Record;
+import org.reactivestreams.Publisher;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.fge.lambdas.Throwing;
+import com.google.common.collect.ImmutableList;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scala.publisher.SMono;
+import scala.Option;
+import scala.collection.JavaConverters;
+import scala.collection.immutable.List;
+import scala.collection.immutable.Seq;
+import scala.jdk.javaapi.OptionConverters;
+import scala.runtime.BoxedUnit;
+
+public class PostgresCustomIdentityDAO implements CustomIdentityDAO {
+    static class Email {
+        private String name;
+        private String email;
```

Comment:
Why don't we get an immutable object? 
We can do a POJO copy if we need to change something in it.
Or a fancy builder?

Reply:
Nope wrong.

Annotate the constructor with `@JsonCreator`, annotate parameters with `@JsonParameter` and use an immutable POJO please.

Reply:
updated. sorry for mistake :))